transformer_engine/pytorch/distributed.py

```diff
@@ -824,3 +833,102 @@ def allreduce(
     handle = torch.distributed.all_reduce(input_, group=tp_group, async_op=async_op)
 
     return input_, handle
+
+
+def _fsdp_scatter_tensors(
+    fsdp_group: dist_group_type,
+    *tensors: torch.Tensor,
+):
+    shapes = []
+    if fsdp_group is not None:
```

Comment:
Does this handle the case where the user specifies the world group with `None`? I see [some logic in FSDP](https://github.com/pytorch/pytorch/blob/b36e390b6c1f4a1a2239ad1e857a2ffd12f49e43/torch/distributed/fsdp/_init_utils.py#L135), but have you confirmed?

Reply:
Yes, I've confirmed this. `prepare_te_modules_for_fsdp(...)` grabs the process group from every FSDP-wrapped TE module's `_FSDPState`, which FSDP always populates with the default process group when the user invokes FSDP with a `None` group. So this group we use in TE can only be `None` when there is no FSDP wrap at all.